### PR TITLE
Search parcel using select within logic

### PR DESCRIPTION
**Pull request by danangmassandy**
This is for https://github.com/kartoza/sawps/issues/1324

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 34 sec

File: django_project/frontend/utils/parcel.py [hunk 1]
```diff
@@ -33,26 +34,41 @@ def find_parcel_base(cls, serialize_cls,
...
     if parcel_keys:
         parcels = parcels.exclude(
             cname__in=parcel_keys
         )
-    parcels = parcels.order_by('cname').distinct('cname')
-    if parcels:
-        cname_list = [a.cname for a in parcels]
+    parcels = parcels.annotate(
+        centroid=Centroid('geom')).order_by('cname').distinct('cname')
+    if parcels.exists():
```
Comment (dimasciput) [inline django_project/frontend/utils/parcel.py:46]: @danangmassandy, the code is nested too deeply. Let's refactor it to make it more readable

**Review by dimasciput — CHANGES_REQUESTED** [review]

File: django_project/frontend/utils/parcel.py (hunk 1 above)
Comment (danangmassandy) [reply django_project/frontend/utils/parcel.py:68]: Updated Dimas,,

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 7 min 3 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 10 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 4 sec

**Pull request merged**